asf_floorplan_interpreter/pipeline/prodigy_to_yolo.py

```diff
@@ -48,43 +54,117 @@ def convert_prod_to_yolo(prod_label, object_to_class_dict):
+
+        # Save the image
+        bucket.upload_fileobj(
+            BytesIO(image_data),
+            os.path.join(output_folder_name, f"images/{data_type}/{image_name}.jpg"),
+        )
+        # Save the labels
+        save_to_s3(
+            BUCKET_NAME,
+            yolo_label,
+            os.path.join(output_folder_name, f"labels/{data_type}/{image_name}.txt"),
+            verbose=False,
+        )
 
 
 if __name__ == "__main__":
-    ## Process the room dataset
+    prodigy_labelled_date = "191023"
+
+    train_prop = 0.6
+    test_prop = 0.2
+    # val_prop = 1 - (train_prop + test_prop) # Dont need to set this
+    if train_prop + test_prop == 1:
+        print("Warning - there will be no data in the validation set")
+
+    print("Process the room dataset")
 
-    file_name = "data/annotation/prodigy_labelled/181023/room_dataset.jsonl"
-    output_folder_name = (
-        "data/annotation/prodigy_labelled/181023/room_dataset_yolo_formatted/"
+    prod_file_name = (
+        f"data/annotation/prodigy_labelled/{prodigy_labelled_date}/room_dataset.jsonl"
+    )
+    yolo_data_folder_name = (
+        f"data/annotation/prodigy_labelled/{prodigy_labelled_date}/room_yolo_formatted/"
     )
     object_to_class_dict = {
         "ROOM": 0,
     }
-    convert_prodigy_file(file_name, object_to_class_dict, output_folder_name)
+    room_yolo_labels = convert_prodigy_file(prod_file_name, object_to_class_dict)
+    split_save_data(room_yolo_labels, train_prop, test_prop, yolo_data_folder_name)
 
-    ## Process the window/door/staircase dataset
+    print("Process the window/door/staircase dataset")
 
-    file_name = "data/annotation/prodigy_labelled/181023/window_door_staircase.jsonl"
-    output_folder_name = (
-        "data/annotation/prodigy_labelled/181023/window_door_staircase_yolo_formatted/"
-    )
+    prod_file_name = f"data/annotation/prodigy_labelled/{prodigy_labelled_date}/window_door_staircase.jsonl"
+    yolo_data_folder_name = f"data/annotation/prodigy_labelled/{prodigy_labelled_date}/window_door_staircase_yolo_formatted/"
     object_to_class_dict = {
         "WINDOW": 0,
         "DOOR": 1,
         "STAIRCASE": 2,
     }
-    convert_prodigy_file(file_name, object_to_class_dict, output_folder_name)
+    window_door_yolo_labels = convert_prodigy_file(prod_file_name, object_to_class_dict)
+    split_save_data(
+        window_door_yolo_labels, train_prop, test_prop, yolo_data_folder_name
+    )
+
+    # Original data was 122 annotations, filtered for accepted labels and deduplicated gives us 105 annotations
+    # Saving 63 image in the training set
+    # Saving 21 image in the test set
+    # Saving 21 image in the val set
+    # Process the window/door/staircase dataset
+    # Original data was 119 annotations, filtered for accepted labels and deduplicated gives us 102 annotations
+    # Saving 61 image in the training set
+    # Saving 20 image in the test set
+    # Saving 21 image in the val set
```

Comment:
This all looks good, but I haven't tested it, because I don't want to overwrite labelled files. Maybe we can talk through quickly next week.